BOM-USAGE.md

```diff
@@ -0,0 +1,187 @@
+# GeoTools BOM Usage Guide
+
+GeoTools 34 introduces the Bill of Materials (BOM) pattern to improve dependency management. This guide explains how to use the BOMs in your projects.
+
+## What is a BOM?
+
+A BOM (Bill of Materials) is a special POM file that manages dependency versions across a project. Rather than specifying versions for each dependency individually, you can import the BOM which provides consistent versioning.
+
+GeoTools provides two BOMs:
+
+1. `gt-platform-dependencies` - Third-party dependency versions
+2. `gt-bom` - GeoTools module versions (also imports platform-dependencies)
+
+## Using the BOMs
+
+### For GeoTools Module Development
+
+When developing GeoTools modules, you don't need to specify versions for internal dependencies:
+
+```xml
+<dependencies>
+  <!-- No version needed for GeoTools modules -->
+  <dependency>
+    <groupId>org.geotools</groupId>
+    <artifactId>gt-main</artifactId>
```

Comment:
Thinking out loud, is there any tool that would enforce not having version numbers in the pom files, to be used in a QA Build? (old habits die hard, and review might not notice if the PR is large)

Reply:
there's not. Yet.

Reply:
Whoa, you had to make a plugin for it ? I was just wondering if it was possible, did not mean to ask for such an amount of development :rofl: 
But thanks!

Reply:
> Aside: I had to move the dependencies from build/maven/pom.xml to build/maven/jjtree-javacc/ and build/maven/xmlcodegen/ since they use the maven 2.0 API. I'd like to upgrade them to 3.8.1 (same we require as minimum in the root pom), but that'd be a separate issue.

@groldan I had a crack at upgrading those Maven plugins in https://github.com/geotools/geotools/pull/3402; but it stranded and I had to abandon